ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestClientImpl.java

```diff
@@ -45,9 +47,33 @@
     @Override
     public IngestionResult ingestFromBlob(BlobSourceInfo blobSourceInfo, IngestionProperties ingestionProperties)
             throws IngestionClientException, IngestionServiceException {
+        return ingestFromBlobImpl(blobSourceInfo, ingestionProperties);
+    }
+
+    @Override
+    public CompletableFuture<IngestionResult> ingestFromBlobAsync(BlobSourceInfo blobSourceInfo, IngestionProperties ingestionProperties) {
+        return CompletableFuture.supplyAsync(
+                () -> {
+                    try {
+                        return ingestFromBlobImpl(blobSourceInfo, ingestionProperties);
+                    } catch (IngestionClientException | IngestionServiceException e) {
+                        log.error("Error when ingestFromBlobAsync()", e);
+                        // Here we throw a CompletionException which extends the RuntimeException.
+                        // the real exception itself would be in the <cause> of this CompletionException
+                        throw new CompletionException(e);
+                        /* We might decide what to return in case of error.
```

Comment:
This comment shouldn't be here. You should make a decision and follow it.
Assuming you want to deal with the ingestion exceptions in the caller function, the way to go is return CompletionException.
In the caller you can do:
catch(CompletionException ex) {
        try {
            throw ex.getCause();
        }
        catch(IngestionClientException | IngestionServiceException e) {
            // do something
        }
      ...
    }